agenta-web/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -9,8 +9,26 @@ const ProtectedRoute: React.FC<PropsWithChildren> = ({children}) => {
     const {pathname} = router
     const [shouldRender, setShouldRender] = useState(false)
     const {isLoading, isProjectId} = useProjectData()
+    const isBusy = useRef(false)
 
     useEffect(() => {
+        const startHandler = (newRoute: string) => {
+            isBusy.current = true
+        }
+        const endHandler = (newRoute: string) => {
+            isBusy.current = false
+        }
+
+        Router.events.on("routeChangeStart", startHandler)
```

Comment:
Small note: Let's use the existing `router` instance from useRouter instead of importing `Router` directly

```suggestion
        router.events.on("routeChangeStart", startHandler)
```